qa-tests-backend/src/test/groovy/DeclarativeConfigTest.groovy

```diff
@@ -301,15 +301,19 @@ splunk:
         assert authProvider
 
         // Verify the groups are created successfully, and specify the origin declarative.
-        def expectedGroups = [VALID_DEFAULT_GROUP, VALID_DECLARATIVE_GROUP]
+        def expectedGroups = [VALID_DECLARATIVE_GROUP, VALID_DEFAULT_GROUP]
+                .sort { it.roleName }
         def groupsResponse = GroupService.getGroups(
                 GroupServiceOuterClass.GetGroupsRequest.newBuilder().setAuthProviderId(authProvider.getId()).build())
-        def expectedProperties = expectedGroups.props
-        verifyAll(groupsResponse.getGroupsList().props) {
-            it.key == expectedProperties.key
-            it.value == expectedProperties.value
-            it.traits.origin == expectedProperties.traits.origin
-            it.authProviderId.every { it == authProvider.id }
+
+        def retrievedGroups = groupsResponse.getGroupsList().collect()
```

Comment:
`collect()` is redundant 
```suggestion
        def retrievedGroups = groupsResponse.getGroupsList()
```

Reply:
Doesn't work without `collect()` since the underlying array is immutable:
```
java.lang.UnsupportedOperationException
|               	at java.base/java.util.Collections$UnmodifiableList.sort(Collections.java:1369)
```